saagieapi/repositories/repositories.py

```diff
@@ -0,0 +1,312 @@
+import logging
+from pathlib import Path
+from typing import Dict, Optional
+
+from gql import gql
+
+from .gql_queries import *
+
+
+class Repositories:
+    def __init__(self, saagie_api):
+        self.saagie_api = saagie_api
+
+    def list(
+        self, minimal: Optional[bool] = False, last_synchronization: bool = True, pprint_result: Optional[bool] = None
+    ) -> Dict:
+        """
+        Get information for all repositories
+        NB: You can only get repositories information if you have the right to
+        access the technology catalog
+
+        Parameters
+        ----------
+        minimal : bool, optional
+            Whether to only return the repository's name and id, default to False
+        last_synchronization : bool, optional
+            Whether to only fetch the last synchronization of each repository
+        pprint_result : bool, optional
+            Whether to pretty print the result of the query, default to
+            saagie_api.pprint_global
+
+        Returns
+        -------
+        dict
+            Dict of repositories
+        """
+        params = {
+            "minimal": minimal,
+            "lastSynchronization": last_synchronization,
+        }
+        query = gql(GQL_LIST_REPOSITORIES)
+        return self.saagie_api.client_gateway.execute(query, variable_values=params, pprint_result=pprint_result)
+
+    def list_minimal(self) -> Dict:
+        """List only repository names, ids and last reversible IDs.
+        NB: You can only list repository if you have at least the viewer role on the
+        catalog
+
+        Returns
+        -------
+        dict
+            Dict of repository ids, names and last reversible IDs
+        """
+        params = {"minimal": True, "lastSynchronization": True}
+
+        return self.saagie_api.client_gateway.execute(query=gql(GQL_LIST_REPOSITORIES), variable_values=params)
```

Comment:
This function is not necessary any more as we can proceed with the function list()

Reply:
Thanks 👍 , it is removed